src/pages/physical-count-itemDetails/index.js

```diff
@@ -0,0 +1,633 @@
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useContext, useEffect, useState, useRef } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Button, Grid } from '@mui/material'
+import { ControlContext } from 'src/providers/ControlContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { SCRepository } from 'src/repositories/SCRepository'
+import FormShell from 'src/components/Shared/FormShell'
+import CustomNumberField from 'src/components/Inputs/CustomNumberField'
+import { useForm } from 'src/hooks/form'
+import * as yup from 'yup'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { SystemChecks } from 'src/resources/SystemChecks'
+import toast from 'react-hot-toast'
+
+const PhysicalCountItemDe = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const [siteStore, setSiteStore] = useState([])
+  const [controllerStore, setControllerStore] = useState([])
+  const [filteredItems, setFilteredItems] = useState([])
+  const [editMode, setEditMode] = useState(false)
+  const [disSkuLookup, setDisSkuLookup] = useState(false)
+  const [jumpToNextLine, setJumpToNextLine] = useState(false)
+  const [showDefaultQty, setShowDefaultQty] = useState(false)
+  const [disableItemDuplicate, setDisableItemDuplicate] = useState(false)
+
+  const { labels: _labels, maxAccess } = useResourceQuery({
+    datasetId: ResourceIds.IVPhysicalCountItemDetails
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      stockCountId: null,
+      siteId: null,
+      controllerId: null,
+      totalQty: null,
+      totalWeight: null,
+      status: 1,
+      SCStatus: null,
+      SCWIP: null,
+      EndofSiteStatus: null,
+      rows: [
+        {
+          id: 1,
+          sku: '',
+          itemId: null,
+          itemName: '',
+          countedQty: 0,
+          weight: 0,
+          metalPurity: 0
+        }
+      ]
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      stockCountId: yup.string().required(),
+      siteId: yup.string().required(),
+      controllerId: yup.string().required(),
+      rows: yup.array().of(
+        yup.object().shape({
+          sku: yup.string().required(),
+          itemName: yup.string().required()
+        })
+      )
+    }),
+    onSubmit: async obj => {
+      const copy = { ...obj }
```

Comment:
If copy is never modified, there’s no need to destructure the obj into copy. You can use obj directly instead.